rllib/utils/actor_manager.py

```diff
@@ -0,0 +1,324 @@
+from collections import defaultdict
+import logging
+from typing import Callable, List, Optional, Tuple, TypeVar
+
+import ray
+from ray.actor import ActorHandle
+from ray.exceptions import RayActorError
+from ray.util.annotations import DeveloperAPI
+
+
+logger = logging.getLogger(__name__)
+# Generic type var for foreach_* methods.
+T = TypeVar("T")
+
+
+class FaultTolerantActorManager:
+    """A manager that is aware of the healthiness states of remote actors.
```

Comment:
nit: delete "states"

Reply:
So will this eventually completely replace the other Actor Manager that we already have and that's being used by e.g. Impala and APPO?